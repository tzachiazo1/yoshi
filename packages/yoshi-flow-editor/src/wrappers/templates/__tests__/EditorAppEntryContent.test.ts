import editorAppEntryContent from '../EditorAppEntryContent';

describe('EditorAppEntry template', () => {
  it('generates correct template for editor app entry', () => {
    const generatedEditorAppEntryContent = editorAppEntryContent({
      editorAppWrapperPath: 'some/editor-app-wrapper',
      componentName: 'MyWidget',
      componentFileName: 'some/components/MyWidget.tsx',
      controllerFileName: 'some/components/controller.ts',
      viewerEntryFileName: 'some/viwer.app.ts',
      sentryConfig: {
        DSN: 'https//dsn.com',
        id: '123',
        projectName: 'some-project',
        teamName: 'some-team',
      },
      experimentsConfig: null,
    });

    expect(generatedEditorAppEntryContent).toMatchSnapshot();
  });

  it('generates correct template for editor app entry without viewerEntryFileName', () => {
    const generatedEditorAppEntryContent = editorAppEntryContent({
      editorAppWrapperPath: 'some/editor-app-wrapper',
      componentName: 'MyWidget',
      componentFileName: 'some/components/MyWidget.tsx',
      controllerFileName: 'some/components/controller.ts',
      viewerEntryFileName: null,
      sentryConfig: null,
      experimentsConfig: null,
    });

    expect(generatedEditorAppEntryContent).toMatchSnapshot();
  });
});
