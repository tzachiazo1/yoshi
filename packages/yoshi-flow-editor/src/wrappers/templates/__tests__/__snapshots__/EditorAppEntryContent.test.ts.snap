// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorAppEntry template generates correct template for editor app entry 1`] = `
"
    import React from 'react';
    import ReactDOM from 'react-dom';
    import EditorAppWrapper from 'some/editor-app-wrapper';

    import UserComponent from 'some/components/MyWidget.tsx';
    import createController from 'some/components/controller.ts';
    
  import * as viewerApp from 'some/viwer.app.ts';
    var importedApp = viewerApp;


    var componentName = 'MyWidget';
    var sentryConfig = {
      DSN: 'https//dsn.com',
      id: '123',
      projectName: 'some-project',
      teamName: 'some-team',
    };

    var experimentsConfig = null;

    var WrappedEditorApp = () => React.createElement(EditorAppWrapper, {
      UserComponent: UserComponent,
      name: componentName,
      sentry: sentryConfig,
      experimentsConfig: experimentsConfig,
      userController: createController,
      customInitAppForPage: importedApp.initAppForPage
    });

    ReactDOM.render(React.createElement(WrappedEditorApp, null), document.getElementById('root'));
"
`;

exports[`EditorAppEntry template generates correct template for editor app entry without viewerEntryFileName 1`] = `
"
    import React from 'react';
    import ReactDOM from 'react-dom';
    import EditorAppWrapper from 'some/editor-app-wrapper';

    import UserComponent from 'some/components/MyWidget.tsx';
    import createController from 'some/components/controller.ts';
    
  var importedApp = {};


    var componentName = 'MyWidget';
    var sentryConfig = null;

    var experimentsConfig = null;

    var WrappedEditorApp = () => React.createElement(EditorAppWrapper, {
      UserComponent: UserComponent,
      name: componentName,
      sentry: sentryConfig,
      experimentsConfig: experimentsConfig,
      userController: createController,
      customInitAppForPage: importedApp.initAppForPage
    });

    ReactDOM.render(React.createElement(WrappedEditorApp, null), document.getElementById('root'));
"
`;
